web/Web/web/CAL/models.py

```diff
@@ -0,0 +1,38 @@
+from config.settings.base import AUTH_USER_MODEL as User
+
+from django.contrib.postgres.fields import JSONField
+from django.db import models
+
+from web.core.models import Session
+
+
+class DS_logging(models.Model):
+    class Meta:
+        unique_together = ['user', 'stratum_num', 'session']
+        index_together = ['user', 'stratum_num', 'session']
+
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    session = models.ForeignKey(Session, on_delete=models.CASCADE)
+
+    stratum_size = models.IntegerField(default=-1)
+    stratum_num = models.IntegerField(default=-1)
+    sample_size = models.IntegerField(default=-1)
+
+    T = models.IntegerField(default=-1)
+    N = models.IntegerField(default=-1)
+    R = models.IntegerField(default=-1)
+
+    sampled_docs = JSONField(null=True, blank=True, default=list)
+    stratum_docs = JSONField(null=True, blank=True, default=list)
+
+    # Todo: Add more fields
+    # created_at = models.DateTimeField(auto_now_add=True, editable=False)
```

Comment:
Uncomment `created_at` and `updated_at` can be deleted.

Reply:
Fixed